base64.js

```diff
@@ -1,6 +1,6 @@
 ;(function () {
 
-  var object = typeof exports != 'undefined' ? exports : self; // #8: web workers
+  var object = typeof exports != 'undefined' ? exports : typeof self != 'undefined' ? self : (1,eval)('this'); // #8: web workers, #31: ExtendScript
```

Comment:
@davidchambers `1,eval` trick works in the illustrator context but not in the photoshop context.  I couldn't get your function working either.  So I went back to the docs https://wwwimages2.adobe.com/content/dam/Adobe/en/devnet/scripting/pdfs/javascript_tools_guide.pdf and it says `$.global` should be available everywhere.

Can you check if `$.global` works for you?  

Also, what should happen if none of the variables are available?  The original code assumed `self` was available